app/actions/page.ts

```diff
@@ -183,32 +184,38 @@ export function changePageDescription(pageId: string, newDescription: string) {
 export function selectPage(
     pageId: string,
     isDrilldown?: boolean,
-    drilldownContext?: any,
+    drilldownContext?: Record<string, any>,
     drilldownPageName?: string
 ): ThunkAction<void, ReduxState, never, AnyAction> {
     return (dispatch, getState) => {
         const location: LocationDescriptorObject = { pathname: `/page/${pageId}` };
-        let newDrilldownContext = getState().drilldownContext || [];
 
         // Clear the widgets data since there is no point in saving data for widgets that are not in view
         dispatch(clearWidgetsData());
 
         // Update context and location depending on page is drilldown
         if (!isDrilldown) {
             dispatch(clearContext());
+            if (drilldownContext) {
+                const newDrilldownContext: DrilldownContext[] = [{ context: drilldownContext }];
+                // eslint-disable-next-line scanjs-rules/assign_to_search
+                location.search = stringify({ c: JSON.stringify(newDrilldownContext) });
+            }
         } else {
             if (!_.isEmpty(drilldownPageName)) {
                 // eslint-disable-next-line scanjs-rules/assign_to_pathname
                 location.pathname += `/${drilldownPageName}`;
             }
+
+            const newDrilldownContext = (getState().drilldownContext || []).slice();
+
             if (drilldownPageName || drilldownContext) {
-                newDrilldownContext = [
-                    ...newDrilldownContext,
-                    {
-                        context: drilldownContext || {},
-                        pageName: drilldownPageName
-                    }
-                ];
+                newDrilldownContext.push({
+                    context: drilldownContext || {},
+                    pageName: drilldownPageName
+                });
+            } else {
+                throw new Error('Drilldown page name and context not provided while doing a drilldown');
```

Comment:
My understanding of this error message is that both drilldown page name and context need to be provided, while the logic above implies that only one of them is required.

Reply:
Hmm... Would

```suggestion
                throw new Error('Either drilldown page name or context have to be provided while doing a drilldown');
```

be better here?

Reply:
Clear to me now 👍.

Reply:
Great :+1: Done